homeassistant/components/control4/config_flow.py

```diff
@@ -0,0 +1,198 @@
+"""Config flow for Control4 integration."""
+from asyncio import TimeoutError as asyncioTimeoutError
+import logging
+
+from aiohttp.client_exceptions import ClientError
+from pyControl4.account import C4Account
+from pyControl4.director import C4Director
+from pyControl4.error_handling import NotFound, Unauthorized
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    CONF_CONTROLLER_NAME,
+    CONF_LIGHT_COLD_START_TRANSITION_TIME,
+    CONF_LIGHT_TRANSITION_TIME,
+    DEFAULT_LIGHT_COLD_START_TRANSITION_TIME,
+    DEFAULT_LIGHT_TRANSITION_TIME,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    MIN_SCAN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured Control4 instances."""
+    return {
+        entry.data.get(CONF_CONTROLLER_NAME)
+        for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class Control4Validator:
+    """Validates that config details can be used to authenticate and communicate with Control4."""
+
+    def __init__(self, host, username, password):
+        """Initialize."""
+        self.host = host
+        self.username = username
+        self.password = password
+        self.account = C4Account(self.username, self.password)
```

Comment:
Please move side effects out of init.  Can you create this in `authenticate` ?
```suggestion
        self.account = None
```